packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -305,14 +305,14 @@ export class DataStores implements IDisposable {
      */
     private makeDataStoreLocallyVisible(id: string): void {
         const localContext = this.contexts.getUnbound(id);
-        assert(!!localContext, 0x15f /* "Could not find unbound context to bind" */);
+        assert(Boolean(localContext), 0x15f /* "Could not find unbound context to bind" */);
 
         /**
          * If the container is not detached, it is globally visible to all clients. This data store should also be
          * globally visible. Move it to attaching state and send an "attach" op for it.
          * If the container is detached, this data store will be part of the summary that makes the container attached.
          */
-        if (this.runtime.attachState !== AttachState.Detached) {
+        if (localContext && this.runtime.attachState !== AttachState.Detached) {
```

Comment:
Above, prefer to assert localContext !== undefined, then you don't need this extra check